manticore/platforms/linux.py

```diff
@@ -1331,7 +1331,43 @@ def sys_rt_sigprocmask(self, cpu, how, newset, oldset):
     def sys_sigprocmask(self, cpu, how, newset, oldset):
         logger.debug("SIGACTION, Ignoring changing signal mask set cmd:%d", how)
         return 0
-
+    
+    def sys_dup(self, fd):
+        '''
+        Duplicates an open file descriptor
+        :rtype: int
+        :param fd: the open file descriptor to duplicate.
+        :return: the new file descriptor.
+        '''
+        
+        if self.files[fd] is None:
```

Comment:
Thank you again for these, really great to see these syscalls finally get added. A few notes on this PR:

1) `fd` might not be in the `files` list and can raise; we made `_get_fd` for those extra checks; check how `lseek` is implemented.